eng/tools/suppressions/test/e2e.test.ts

```diff
@@ -0,0 +1,38 @@
+import { expect, test } from "vitest";
+import { getSuppressions } from "../src/suppressions.js";
+import { join } from "path";
+
+async function getTestSuppressions(path: string, tool: string = "TestTool") {
+  return await getSuppressions(tool, join(__dirname, "e2e", path));
+}
+
+test.concurrent("no suppressions.yaml", async () => {
```

Comment:
Would you be interested in using in-memory file system? Like e.g. https://www.npmjs.com/package/mock-fs?

What I found when writing tests is that sometimes I need many small "file snippets" or even I produce file contents programmatically. Then going through the file system seems wasteful - either there will be many small files, or the code will have to write out files to file system just to write them back moment later.

One could make an argument that the code logic should be able to read in-memory string so that file mocking should not be necessary - if so, then we may consider having only very few e2e tests that actually touch the file system (to test it can actually read and write files), and make all the other tests operate on string inputs, and avoid file fixtures completely. But on the other hand one could say that having files with test data present in the file system makes it easier to inspect/debug things.

In any case, basically I see 3 options:
- files with test data in file system
- mocked in-memory file system
- string input instead of file input (whether the file is actual or in-memory mock)

and I think we could have a hybrid test suite potentially leveraging all these 3 approaches, if we end up having good use cases for them.

Reply:
I'm fine allowing a hybrid of all 3.

I think "files with test data in file system" is the best for a reasonable number of tests (relative to the size of the source code).  Easy to setup, run, debug, etc.  Downside is it add mores files to git, but if they are small plaintext files that change rarely, the impact should be minimal.

I think "string input" instead of "file input" is like "unit test" vs "e2e test", so both can make sense.  Use string input if you are only testing an algorithm from string -> output.

Personally I don't like mocking outside exceptional situations, where the other approaches wouldn't scale or are impossible.

I'm currently using a hybrid of "files in filesystem" and "string input" to test suppressions.  Partially because I started with "string input" and don't want to rewrite them.  If I were starting from scratch, I might only use "files in filesystem".  The tool is small enough, it just needs reasonable testing, nothing comprehensive.

Reply:
If the test code itself needs to generate content at runtime, then I agree, writing to disk seems wasteful (though if the test is reliable and fast, who cares).  In this case, I'd recommend using some in-memory solution, preferably more "string input" and less "mocking".